lib/cfn-model/validator/cloudformation_validator.rb

```diff
@@ -1,12 +1,30 @@
 require_relative 'schema_generator'
 require 'kwalify'
+require 'json'
 
 class CloudFormationValidator
   def validate(cloudformation_string)
-    schema = SchemaGenerator.new.generate cloudformation_string
+    if json_text?(cloudformation_string)
+      raise ParserError.new('Invalid JSON!') unless valid_json?(cloudformation_string)
+    end
 
+    schema = SchemaGenerator.new.generate cloudformation_string
     validator = Kwalify::Validator.new(schema)
-
     validator.validate(YAML.load(cloudformation_string))
   end
+
+  private
+
+  def json_text?(cloudformation_string)
+    first_character = cloudformation_string.gsub(/\s/, '').split('').first
+    matches = cloudformation_string.scan(/\{[^}]*\}/)
+    first_character == '{' && !matches.empty?
```

Comment:
scratching my chin - technically json could start with [, but i'm not sure we care vs just having it fall through to yaml parsing and raise an error anyways.

can we get a test(s) for proving how this works against YAML docs? and especially push it on hybrid docs that are YAML with JSON